### PR TITLE
.Net: instead running two queries now a single query can do the same thing

### PR DESCRIPTION
**Pull request by atiq-bs23**
### Motivation and Context
1. Instead of executing two SQL queries, a single query can do the same thing.
2. Performance improvement

### Description
Previously the **InternalUpsertAsync** method was executing two sql query by calling two individual methods - First **UpdateAsync** and then second **InsertOrIgnoreAsync**.

The new method uses the INSERT OR REPLACE statement to either update an existing row or insert a new one based on the unique collection and key combination

### Contribution Checklist

- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [ ] All unit tests pass, same unit test can detect the bug
- [ ] I didn't break anyone :smile:

### CODE REVIEW

**Comment by atiq-bs23** [comment]
@microsoft-github-policy-service agree

**Review by RogerBarreto — APPROVED** [review]

**Comment by atiq-bs23** [comment]
@RogerBarreto should I remove the **InsertOrIgnoreAsync** method as there are no other reference where this method used.
I didn't remove this method in the current commit.

File: dotnet/src/Connectors/Connectors.Memory.Sqlite/Database.cs [hunk 1]
```diff
@@ -73,6 +73,21 @@ public async Task CreateCollectionAsync(SqliteConnection conn, string collection
...
+        VALUES(@collection, @key, @metadata, @embedding, @timestamp);";
+        cmd.Parameters.AddWithValue("@collection", collection);
+        cmd.Parameters.AddWithValue("@key", key);
+        cmd.Parameters.AddWithValue("@metadata", metadata ?? string.Empty);
+        cmd.Parameters.AddWithValue("@embedding", embedding ?? string.Empty);
+        cmd.Parameters.AddWithValue("@timestamp", timestamp ?? string.Empty);
+        await cmd.ExecuteNonQueryAsync(cancellationToken).ConfigureAwait(false);
+    }
+
     public async Task InsertOrIgnoreAsync(SqliteConnection conn,
```
Comment (westey-m) [inline dotnet/src/Connectors/Connectors.Memory.Sqlite/Database.cs:91]: @atiq-bs23 as you suggested, yes, please remove this method now that it's not required anymore.  Looks like UpdateAsync can be removed as well actually.

Also, thanks for contributing this improvement!

Comment (atiq-bs23) [reply dotnet/src/Connectors/Connectors.Memory.Sqlite/Database.cs:91]: Okay, removing

Comment (atiq-bs23) [reply dotnet/src/Connectors/Connectors.Memory.Sqlite/Database.cs]: Removed @westey-m

**Review by westey-m — APPROVED** [review]

**Pull request merged**